pkg/gpu/ebpf/c/runtime/gpu.c

```diff
@@ -168,4 +169,49 @@ int BPF_URETPROBE(uretprobe__cudaStreamSynchronize) {
     return 0;
 }
 
+SEC("uprobe/cudaSetDevice")
+int BPF_UPROBE(uprobe__cudaSetDevice, int device) {
+    __u64 pid_tgid = bpf_get_current_pid_tgid();
+
+    log_debug("cudaSetDevice: pid_tgid=%llu, device=%u", pid_tgid, device);
```

Comment:
it will affect others that will try to debug their pieces of code. we don't have written guidelines, but i think we should  avoid log messages that potentially will emit a lot of lines in the log, even in debug level

Reply:
This would only emit when the CUDA library is being used and the gpu-monitoring module enabled, and even then it will just write one-two log lines for each `cudaSetDevice` call (which is probably going to happen a handful of times per process, as it's a per-thread global setting). I don't think we gain much by removing it.